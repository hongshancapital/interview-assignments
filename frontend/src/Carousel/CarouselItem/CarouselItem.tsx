import { FC } from 'react';
import styles from './CarouselItem.module.scss';

export type CarouselItemProps = {
  imgSrc: string;
  title: string;
  subTitle?: string;
  className?: string;
};

export const CarouselItem: FC<CarouselItemProps> = ({
  imgSrc,
  title,
  subTitle = '',
  className = '',
}) => {
  return (
    <div
      className={`${styles.carouselItem} ${className}`}
      aria-label="carousel item"
    >
      <h2 className={styles.carouselItemTitle}>{title}</h2>
      <h3 className={styles.carouselItemSubTitle}>{subTitle}</h3>
      <img className={styles.carouselItemImage} src={imgSrc} alt={title} />
    </div>
  );
};
