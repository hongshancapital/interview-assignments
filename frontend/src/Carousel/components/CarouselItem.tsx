import React from 'react';

const CarouselItem: React.FC<{ width: number; children: React.ReactNode }> = ({ width, children }) => {
  return <div style={{ width }} className="carousel-item">{children}</div>
}

if (process.env.NODE_ENV !== 'production') {
  CarouselItem.displayName = 'CarouselItem';
}

export default React.memo(CarouselItem);