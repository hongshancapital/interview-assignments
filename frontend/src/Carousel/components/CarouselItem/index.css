.carousel-item {
    position: relative;
    width: 100vw;
    height: 100vh;
}
.product-img {
    width: 100vw;
    height: 100vh;
    /* 图片的fit没有找到很好的方式，cover会导致产品图标有点大 */
    /* todo: 如果要完全符合，需要切图 */
    object-fit: cover;
}

.title {
    position: absolute;
    left: 50%;
    top: 20%;
    transform: translateX(-50%);
}

h1 {
    font-size: 60px;
}

h3 {
    font-size: 30px;
}