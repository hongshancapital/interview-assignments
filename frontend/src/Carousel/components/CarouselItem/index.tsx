import React from 'react';
import { ICarouselItemProps } from '../../index';
import './index.css';

export const CarouselItem = ({
    color,
    title,
    subTitle,
    imageName
}: ICarouselItemProps) => {
    return (
        <div className={`carousel-item`}>
            <img
                className='product-img'
                src={imageName}
                alt="product"
            />
            <div className='title'>
                {
                    title.map((item: string, index: number) => {
                        return <h1 style={{color}} key={index}>{item}</h1>
                    })
                }
                {
                    (subTitle || []).map((item: string, index: number) => {
                        return <h3 style={{color}} key={index}>{item}</h3>
                    })
                }
            </div>
        </div>
    );
};