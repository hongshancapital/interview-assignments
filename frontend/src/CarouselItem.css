.carousel-item-container {
  position: relative;
  width: 100%;
  height: 100vh;
}

.content {
  position: absolute;
  top: 25%;
  width: 100%;
  text-align: center;
  z-index: 1;
}

.image {
  width: 50%;
  height: 50%;
  background-position: center bottom;
  background-size: cover;
  position: absolute;
  top: 38%;
  left: 50%;
  transform: translateX(-50%);
}
