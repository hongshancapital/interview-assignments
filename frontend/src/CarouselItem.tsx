import React from 'react';
import './CarouselItem.css';

interface Props {
	imageUrl: string;
	itemStyle: React.CSSProperties;
	titles?: string[];
	texts?: string[];
}

export default function CarouselItem(props: Props) {
	const { imageUrl, itemStyle, titles, texts } = props;
	return (
		<div className="carousel-item-container" style={itemStyle}>
			<div className="content">
				{titles &&
					titles.map((title, index) => (
						<div className="title" key={index}>
							{title}
						</div>
					))}
				{texts && (
					<div className="text">
						{texts.map((text, index) => (
							<p key={index}>{text}</p>
						))}
					</div>
				)}
			</div>
			<div
				className="image"
				style={{
					backgroundImage: `url(${imageUrl})`,
				}}
			></div>
		</div>
	);
}
