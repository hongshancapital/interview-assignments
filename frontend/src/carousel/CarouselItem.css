.scroll_item {
  color: #fff;
  background-color: #111111;
  padding-top: 130px;
}
.scroll_item .title {
  font-size: 36px;
}
.scroll_item .desc {
  font-size: 26px;
}
.scroll_item_1 {
  background: url('../assets/iphone.png') center 245px no-repeat;
  background-size: 600px;
  background-color: #111111;
}
.scroll_item_2 {
  color: #000;
  background: url('../assets/tablet.png') center 245px no-repeat;
  background-size: 1200px;
  background-color: #f9f9f9;
}
.scroll_item_3 {
  color: #000;
  background: url('../assets/airpods.png') center 130px no-repeat;
  background-size: 2400px;
  background-color: #f0f0f0;
}
