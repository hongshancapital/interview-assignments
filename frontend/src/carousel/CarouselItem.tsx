import React from 'react';
import './CarouselItem.css'

// 轮播 item
const SwipeItem: React.FC = () => (
  <React.Fragment>
    <div className="scroll_item scroll_item_1">
      <h2 className='title'>xPhone</h2>
      <section className='desc'>Lots to love. Less to spend <br /> Starting at $399.</section>
    </div>
    <div className="scroll_item scroll_item_2">
    <h2 className='title'>Tablet</h2>
      <section className='desc'>Just the right amount of everything.</section>
    </div>
    <div className="scroll_item scroll_item_3">
    <section className='desc'>Buy a Tablet or xPhone for college. <br />Get arPods.</section>
    </div>
  </React.Fragment>
)

export default SwipeItem
