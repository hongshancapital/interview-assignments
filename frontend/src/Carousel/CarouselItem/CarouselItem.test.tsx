import { render, screen } from '@testing-library/react';
import { CarouselItem } from './CarouselItem';

describe('CarouselItem', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <CarouselItem
        imgSrc="testImage"
        title="test title"
        subTitle="test subtitle"
        className="testClass"
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('does not render optional props if not provided', () => {
    render(<CarouselItem imgSrc="testImage" title="test title" />);

    expect(screen.getByRole('heading', { level: 3 })).toBeEmptyDOMElement();
    expect(screen.getByLabelText('carousel item')).toHaveClass('carouselItem ');
  });
});
