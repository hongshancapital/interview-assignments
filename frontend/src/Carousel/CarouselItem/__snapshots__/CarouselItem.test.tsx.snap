// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselItem renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-label="carousel item"
    class="carouselItem testClass"
  >
    <h2
      class="carouselItemTitle"
    >
      test title
    </h2>
    <h3
      class="carouselItemSubTitle"
    >
      test subtitle
    </h3>
    <img
      alt="test title"
      class="carouselItemImage"
      src="testImage"
    />
  </div>
</DocumentFragment>
`;
